Makefile

```diff
@@ -13,53 +19,48 @@ build-all: build-darwin build-linux
 
 # Specific linux build target, making it easy to work with the docker acceptance
 # tests on OSX
-bin/%.linux_amd64:
+bin/%.linux_amd64: vet
```

Comment:
I'd really prefer if we don't force people to run vet in their dev flows- our preference in the past has been to enforce things in CI, but allow people to opt into linting locally.

Is there any reason we'd want to change?

Reply:
Only that vet can detect errors the compiler can't. This is also a kubebuilder v2 change (as in this is what they do in their generated Makefile. We don't need to follow practice and we could add it as a CI step.

Reply:
So running vet locally is a great idea, but enforcing that you run vet on every compilation loop means you apply your opinions to another person's setup, making the choice for them.

Running it in CI means we don't merge anything that vet would fail, because you'll go red on your PR. Anyone who wants to can run vet locally before they commit, so that's easy. And everyone who just wants to build the binary is able to build the binary without also running vet, which is good.

Reply:
I will add a vet step to CI instead 👍